### PR TITLE
Add Bazel Vision document.

### PR DESCRIPTION
**Pull request by dslomov**
Add Bazel Vision document.

### CODE REVIEW

File: site/docs/bazel-vision.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+1. Bazel provides a **uniform, extensible language** to define builds for any language
+   or platform.
+1. Bazel allows your builds **to scale** by connecting to remote execution and caching services.
+1. Bazel works across **all major development platforms** (Linux, MacOS, and Windows).
+1. We accept that adopting Bazel requires effort, but **gradual adoption** is possible.
+   Bazel interfaces with de-facto standard tools for a given language/platform.
+
+## Serving language communities
+
+Software engineering evolves in the context of language communities — typically,
```
Comment (jerrymarino) [inline site/docs/bazel-vision.md]: As someone applying Bazel to Apple based development, this really resonates! It seems like Bazel provides first class support for languages, and the idioms used by those communities. The pathological case is when the requirements/needs of the community conflict with what Bazel solves for and prioritizes. ( i.e. the `#include` style preferences of Bazel  https://github.com/bazelbuild/bazel/pull/3712 ), so it is good to see this addressed  🎉

File: site/docs/bazel-vision.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+self-organizing groups of people who use common tools and practices. To be of use to members
+of a language community, high-quality Bazel rules must be available that integrate with the
+workflows and conventions of that community. Bazel is committed to be extensible and open,
+and to support good rulesets for any language.
```
Comment (jerrymarino) [inline site/docs/bazel-vision.md]: nitpick: consider replacing `good` with `idomatic`.

**Comment by spomorski** [comment]
Did you mean "idiomatic"?

**Comment by jerrymarino** [comment]
@spomorski yes 👍

**Comment by dslomov** [comment]
@spomorski ping for review

**Pull request closed without merging**